tools/rostopic/src/rostopic/__init__.py

```diff
@@ -297,7 +297,7 @@ def msgevalgen(pattern):
     for f in fields:
         if '[' in f:
             field_name, rest = f.split('[', 1)
-            if rest[-1] != ']':
+            if not rest or rest[-1] != ']':
                 print("missing closing ']' in slice spec '%s'" % f, file=sys.stderr)
```

Comment:
This error message should also include the possibility that the brackets are just empty, not just that a trailing `]` is missing.

Also, I missed this last time, it would be more idiomatic Python to do `not rest.endswith(']')` rather than `rest[-1] != ']'`.

One more thought is that it might be good to include a test for something like `/vals[/foo` and/or `/vals[]/foo`.